tests/test_lit_server.py

```diff
@@ -84,3 +92,28 @@ def test_single_loop(simple_litapi, loop_args):
 
     with pytest.raises(StopIteration, match="exit loop"):
         run_single_loop(lit_api_mock, requests_queue, request_buffer)
+
+
+def test_generate_client_file(capsys, lit_server):
+    lit_server.generate_client_file()
+    src_path = os.path.join(os.path.dirname(server.__file__), "python_client.py")
+
+    dest_path = os.path.join(os.getcwd(), "client.py")
+    stmt = f"File '{src_path}' copied to '{dest_path}'\n"
+
+    captured = capsys.readouterr()
+    assert captured.out == stmt
+
+    assert os.path.exists(dest_path)
+    assert lit_server.generate_client_file() is None
+    os.remove(dest_path)
+
+
+def test_run(lit_server):
+    t = Thread(target=lit_server.run, daemon=True)
+    t.start()
+    time.sleep(2)
+    assert os.path.exists("client.py")
+    output = subprocess.run("python client.py", shell=True, capture_output=True, text=True).stdout
+    assert '{"output":16.0}' in output
+    os.remove("client.py")
```

Comment:
this test is incorrect. it needs to run the server from the command line, then run the client from the command line to assert it gets the response

Reply:
updated the code to run server from command line @williamFalcon